src/workerd/io/io-context.c++

```diff
@@ -429,6 +431,11 @@ void IoContext::addWaitUntil(kj::Promise<void> promise) {
     }
   }
 
+  if (!incomingRequests.empty()) {
+    KJ_LOG(WARNING, "Adding task to IoContext with no current IncomingRequest",
```

Comment:
Changed to a warning since, if my hypothesis is correct, making it an assert might break prod otherwise.